charon/tests/ui/arrays.out

```diff
@@ -1747,19 +1737,17 @@ fn test_crate::slice_pattern_4<'_0>(@1: &'_0 (Slice<()>))
     let @3: usize; // anonymous local
     let @4: bool; // anonymous local
     let _named@5: &'_ (()); // local
-    let @6: usize; // anonymous local
-    let @7: &'_ (Slice<()>); // anonymous local
-    let @8: &'_ (()); // anonymous local
+    let @6: &'_ (Slice<()>); // anonymous local
+    let @7: &'_ (()); // anonymous local
 
     @fake_read(x@1)
     @2 := len(*(x@1))
     @3 := const (1 : usize)
     @4 := move (@2) == move (@3)
     if move (@4) {
-        @6 := const (0 : usize)
-        @7 := &*(x@1)
-        @8 := @SliceIndexShared<'_, ()>(move (@7), copy (@6))
-        _named@5 := &*(@8)
+        @6 := &*(x@1)
+        @7 := @SliceIndexShared<'_, ()>(move (@6), const (0 : usize))
+        _named@5 := &*(@7)
```

Comment:
Ah, we have a `simplify_constants` pass that moves constant operands into separate locals but unfortunately it runs before `index_to_function_calls`. That's why there's a change here; that's not great. Unfortunately `simplify_constants` operates on ullbc and `index_to_function_calls` operates on llbc so we can't easily reorder them... We might have to rewrite `simplify_constants` to operate on llbc, it's not the first time I've wanted that.

Reply:
Oops I'm pointing at the wrong thing